File: AdventJS/ejercicio_6.js
function inBox(box){
    box.forEach(subarray => {
        if (subarray[0] == '*' || subarray[subarray.length()-1] == '*'){
            return false
        }
    });

    var box_plano = box.flat();
    if(box_plano.includes('*')){
        return true;
    }else{
        return false;
    }
}

// Primer ejercicio con js

function prepareGifts(gifts) {
    gifts = gifts.filter((valor, arreglo, index) => {
        return arreglo.index(valor) === index
    })
}

gifts = [1,2,3,7,7,8,9,9,0,1];
console.log(prepareGifts(gifts))

File: AdventJS/ejercicio_6.py
class Regalo:
    def in_box(self, box):
        for subarray in box:
            if subarray[0] == '*' or subarray[(len(subarray)-1)] == '*':
                return False
        
        box_plano = [elemento for sublista in box[1:-1] for elemento in sublista]
        if box_plano.count('*') >= 1:
            return True
        else: 
            return False


box = ([
  "###",
  "#*#",
  "###"])
obj = Regalo()
print(obj.in_box(box))
